src/main/java/life/qbic/portal/portlet/ImageInfo.java

```diff
@@ -0,0 +1,54 @@
+package life.qbic.portal.portlet;
+
+/**
+ * <short description>
+ * <p>
+ * <detailed description>
+ *
+ */
+public class ImageInfo {
+  private final long imageId;
+  private final String name;
+  private final byte[] thumbnail;
+  private final String size;
+  private final String timePoints;
+  private final String channels;
+
+  private ImageInfo() {
+    throw new AssertionError("Default constructor of " + ImageInfo.class + " is disabled.");
+  }
+
+  public ImageInfo(long imageId, String name, byte[] thumbnail, String size,
+      String timePoints, String channels) {
+    this.imageId = imageId;
+    this.name = name;
+    this.thumbnail = thumbnail;
+    this.size = size;
+    this.timePoints = timePoints;
+    this.channels = channels;
+  }
+
+  public long getImageId() {
+    return imageId;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public byte[] getThumbnail() {
+    return thumbnail;
```

Comment:
you are returning a reference to the thumbnail array here, which means that you cannot guarantee the immutable state of this class any more. I suggest to return a deep copy of the byte array.

Reply:
I overlooked that, thank you

Reply:
73d2a13